src/Elcodi/Component/User/Entity/Abstracts/AbstractUser.php

```diff
@@ -311,10 +312,16 @@ public function getFullName()
      */
     public function setPassword($password)
     {
-        if (null !== $password) {
-            $this->password = $password;
+        if (null === $password) {
+            return $this;
         }
 
+        if (!is_string($password) || empty($password)) {
```

Comment:
what about a `trim` instead of `empty`? this change should disallow passwords made of white spaces